Add UUencode

malathit opened the pull request:
Closes #433

malathit committed:
Create uuencode.py

malathit commented:
```
ciphey -m ciphey/basemods/Decoders/uuencode.py -t "begin 666 <data>
> %:&5L;&\ 
>  
> end
>  "
Result 'hello' (y/N): y
Format used:
  uuencode
Final result: "hello"
```

bee-san commented:
Hey! We need tests for this :D 

https://github.com/Ciphey/Ciphey/wiki/Testing-your-cracker---decoder

SkeletalDemise commented:
Hello. This doesn't seem to work for me. I tried putting ````begin 644 dcode_uuencode
05&AI<R!I<R!A('-E8W)E=```
`
end```` in a file and giving it to Ciphey, but it wouldn't decode. 

I also tried just doing ````ciphey -t "05&AI<R!I<R!A('-E8W)E=```"```` but it also wouldn't work. Can you make sure your decoder works?

bee-san commented:
> Hello. This doesn't seem to work for me. I tried putting `` begin 644 dcode_uuencode 05&AI<R!I<R!A('-E8W)E=``` ` end `` in a file and giving it to Ciphey, but it wouldn't decode.
> 
> I also tried just doing `` ciphey -t "05&AI<R!I<R!A('-E8W)E=```" `` but it also wouldn't work. Can you make sure your decoder works?

Ah, that's because it's not actually hooked up to Ciphey. It's not in `init`. 

Note:

> ⚠️ Warning: This is a step often overlooked. In the Decoding folder there is a file called __init__.py. Edit this file and add your decoder to it.

In https://github.com/Ciphey/Ciphey/wiki/Adding-your-own-ciphers

Please also add your decoder to `init` :D

SkeletalDemise commented:
Yes you should add it to init.py, but that's not why it doesn't work for me. I added it locally and tested it, and it doesn't work for some reason.

malathit committed:
add to init & adding test

malathit committed:
Merge branch 'master' of https://github.com/Ciphey/Ciphey into patch-2

malathit committed:
removing tests

malathit committed:
adding test again

malathit commented:
> Yes you should add it to init.py, but that's not why it doesn't work for me. I added it locally and tested it, and it doesn't work for some reason.

I have added a test case and in the test case if you see, the encoded text it has new lines(\n). I think the new lines are missing in your input and that is why the decoder was not working. This is how I encoded the input string
```
cat /tmp/hello.txt 
Hello my name is bee and I like dog and apple and tree
```
```
uuencode  -o /tmp/hello1.txt /tmp/hello.txt /dev/stdout
```
```
cat /tmp/hello1.txt
begin 644 /dev/stdout
M2&5L;&\@;7D@;F%M92!I<R!B964@86YD($D@;&EK92!D;V<@86YD(&%P<&QE
)(&%N9"!T<F5E
`
end

```
I tested like this
```
ciphey -m uuencode.py -f /tmp/hello1.txt
Result 'Hello my name is bee and I like dog and apple and tree' (y/N): y
Format used:
  uuencode
Final result: "Hello my name is bee and I like dog and apple and tree"
```

malathit committed:
formatting and removed unwanted code

malathit committed:
Merge branch 'master' into patch-2

malathit committed:
skip decoding if input not starts with 'begin'

malathit committed:
Merge branch 'patch-2' of https://github.com/malathit/Ciphey into patch-2

malathit committed:
support for decode without the begin prefix and end suffix

malathit commented:
As discussed with @SkeletalDemise, added the support to decode strings without the `begin` prefix and `end` suffix, using the package `binascii`

malathit committed:
Merge branch 'master' of https://github.com/Ciphey/Ciphey into patch-2

bee-san commented:
Is it possible to have both newlines and no newlines to work? :) How common is new lines in UUencode?

malathit commented:
> Is it possible to have both newlines and no newlines to work? :) How common is new lines in UUencode?

@bee-san looks like the uuencode algorithm encodes data in 45 bytes chunk. If the input length is more than 45 bytes, the algorithm will split the data in 45 bytes chunk and then output each of the 45 bytes chunk in a line. So I guess the new lines are common. 

As of now, the code supports input with or without new lines. The only thing though is if the original text is >45 bytes and the input text to ciphey doesn't have the new line, then the code may not work. It will work with a new line though as given below

```
malathi@my-system > ciphey -m uuencode.py -t 'M2&5L;&\@;7D@;F%M92!I<R!B964@86YD($D@;&EK92!D;V<@86YD(&%P<&QE              
quote> )(&%N9"!T<F5E
quote> '
Result 'Hello my name is bee and I like dog and apple and tree' (y/N): y
Format used:
  uuencode
Final result: "Hello my name is bee and I like dog and apple and tree"
```

SkeletalDemise committed:
Clean up some things

bee-san commented:
Hey we're changing from MIT to GPLv3 because some of our projects are licensed with that, do you approve? If you do not approve we will be forced to delete your code and rewrite it 😢 

By approving of this change, you are agreeing to re-license the code you wrote under GPLv3.

malathit commented:
I approve of this change

On Thu, 20 May 2021 at 16:02, Autumn ***@***.***> wrote:

> Hey we're changing from MIT to GPLv3 because some of our projects are
> licensed with that, do you approve? If you do not approve we will be forced
> to delete your code and rewrite it 😢
>
> By approving of this change, you are agreeing to re-license the code you
> wrote under GPLv3.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ciphey/Ciphey/pull/467#issuecomment-844964513>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUPJ5FUFZYMPG56YEK4HOLTOTQMVANCNFSM4SCXXWQQ>
> .
>


-- 

Regards,
Malathi
